### PR TITLE
Contest History Screen

### PR DESCRIPTION
**Pull request by Fouad-sys**
This pull request adds navigation to the list of past and current Tournaments Screen with a button to create a new Tournament.

### CODE REVIEW

**Review by alexandrepiveteau — CHANGES_REQUESTED** [review]

File: mobile/src/main/java/ch/epfl/sdp/mobile/ui/tournaments/ContestScreen.kt [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+fun <C : Contest> ContestScreen(
+    state: ContestScreenState<C>,
+    modifier: Modifier = Modifier,
+    key: ((C) -> Any)? = null,
+    contentPadding: PaddingValues = PaddingValues()
+) {
+
+  Scaffold(
+      topBar = { TopAppBar() },
+      modifier = modifier.padding(contentPadding),
```
Comment (alexandrepiveteau) [inline mobile/src/main/java/ch/epfl/sdp/mobile/ui/tournaments/ContestScreen.kt:38]: You're applying the padding twice : once in the `Scaffold`, and once in the `LazyColumn`.

File: mobile/src/main/java/ch/epfl/sdp/mobile/ui/tournaments/ContestScreen.kt [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+ *
+ * @param onClick callback function called when the button is clicked on.
+ * @param modifier the [Modifier] for this composable.
+ */
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+fun NewContestButton(onClick: () -> Unit, modifier: Modifier = Modifier) {
+  val strings = LocalLocalizedStrings.current
+
+  FloatingActionButton(
```
Comment (alexandrepiveteau) [inline mobile/src/main/java/ch/epfl/sdp/mobile/ui/tournaments/ContestScreen.kt:81]: Please use `ExtendedFloatingActionButton` instead.

File: mobile/src/main/java/ch/epfl/sdp/mobile/state/StatefulContestsScreen.kt [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+import kotlin.time.Duration.Companion.hours
+
+/** An adapter that is of type [Contest] and contains the uid. */
+data class ContestAdapter(
+    val uid: String,
+    override val name: String,
+    override val creationTime: Duration,
+    override val personStatus: BadgeType,
+    override val status: ContestStatus
+) : Contest {}
```
Comment (alexandrepiveteau) [inline mobile/src/main/java/ch/epfl/sdp/mobile/state/StatefulContestsScreen.kt:20]: I don't think the `{}` are necessary.

File: mobile/src/main/java/ch/epfl/sdp/mobile/state/StatefulContestsScreen.kt [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+/**
+ * An implementation of the [ContestScreenState] that performs a given profile's [Contest] requests.
+ */
+class TournamentScreenState() : ContestScreenState<ContestAdapter> {
+
+  // override var contests by mutableStateOf(emptyList<Contest>())
+  // private set
```
Comment (alexandrepiveteau) [inline mobile/src/main/java/ch/epfl/sdp/mobile/state/StatefulContestsScreen.kt:28]:
```suggestion
```

File: mobile/src/main/java/ch/epfl/sdp/mobile/ui/tournaments/Contest.kt [hunk 5]
```diff
@@ -0,0 +1,18 @@
+package ch.epfl.sdp.mobile.ui.tournaments
+
+import kotlin.time.Duration
+
+/** The button to create a new tournament. */
+interface Contest {
```
Comment (alexandrepiveteau) [inline mobile/src/main/java/ch/epfl/sdp/mobile/ui/tournaments/Contest.kt:6]: I would name this `ContestInfo`, to better distinguish from composables which might be named `Contest`.

File: mobile/src/main/java/ch/epfl/sdp/mobile/ui/tournaments/ContestItem.kt [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+        Text(
+            text = title,
+            color = PawniesColors.Green200,
+            style = MaterialTheme.typography.subtitle2,
+            modifier = modifier)
+      },
+      trailing = {
+        Row(verticalAlignment = Alignment.CenterVertically) {
+          Badge(contest.personStatus, onClick, modifier)
+        }
```
Comment (alexandrepiveteau) [inline mobile/src/main/java/ch/epfl/sdp/mobile/ui/tournaments/ContestItem.kt:47]: You can put the Badge in the lambda directly. A `Row` with a single child will simply wrap it.

File: mobile/src/main/java/ch/epfl/sdp/mobile/ui/tournaments/ContestItem.kt [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+ */
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+fun ContestItem(contest: Contest, onClick: () -> Unit, modifier: Modifier = Modifier) {
+  val strings = LocalLocalizedStrings.current
+  val title = displayContestStatus(strings, contest)
+
+  ListItem(
+      modifier = Modifier.clickable { onClick() },
+      text = { Text(contest.name, color = PawniesColors.Green500) },
```
Comment (alexandrepiveteau) [inline mobile/src/main/java/ch/epfl/sdp/mobile/ui/tournaments/ContestItem.kt:36]: You should set the font for this text.

**Review by alexandrepiveteau — CHANGES_REQUESTED** [review]

File: mobile/src/main/java/ch/epfl/sdp/mobile/state/StatefulContestsScreen.kt [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+import androidx.compose.ui.Modifier
+import ch.epfl.sdp.mobile.ui.tournaments.*
+import ch.epfl.sdp.mobile.ui.tournaments.Status.*
+import ch.epfl.sdp.mobile.ui.tournaments.Status.ContestStatus.*
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.days
+import kotlin.time.Duration.Companion.hours
+
+/** An adapter that is of type [Contest] and contains the uid. */
+data class ContestAdapter(
```
Comment (KurohanaJuri) [inline mobile/src/main/java/ch/epfl/sdp/mobile/state/StatefulContestsScreen.kt:14]: Missing parameters in the documentation

File: mobile/src/main/java/ch/epfl/sdp/mobile/state/StatefulContestsScreen.kt [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+ */
+class TournamentScreenState() : ContestScreenState<ContestAdapter> {
+
+  // override var contests by mutableStateOf(emptyList<Contest>())
+  // private set
+  override val contests =
+      listOf(
+          createContest("1", "EPFL Grand Prix", 1.days, ONGOING, BadgeType.Admin),
+          createContest("2", "Pawn Party", 2.days, DONE, BadgeType.Participant),
+          createContest("3", "Never gonna chess", 3.hours, ONGOING, BadgeType.Join))
```
Comment (KurohanaJuri) [inline mobile/src/main/java/ch/epfl/sdp/mobile/state/StatefulContestsScreen.kt:33]: When you create a dummy data like this, add To-do keyword to notify to other teammate that it is a temporary value

File: mobile/src/main/java/ch/epfl/sdp/mobile/state/StatefulContestsScreen.kt [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+  override fun onNewContestClick() {}
+  override fun onContestClick(C: Contest) {}
+}
+
+private fun createContest(
```
Comment (KurohanaJuri) [inline mobile/src/main/java/ch/epfl/sdp/mobile/state/StatefulContestsScreen.kt:38]: Missing documentation

**Comment by alexandrepiveteau** [comment]
Included in #327.

**Pull request closed without merging**